components/brave_shields/browser/domain_blocked_navigation_throttle.cc

```diff
@@ -0,0 +1,165 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_shields/browser/domain_blocked_navigation_throttle.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/bind.h"
+#include "base/task/post_task.h"
+#include "base/threading/thread_task_runner_handle.h"
+#include "brave/components/brave_shields/browser/ad_block_custom_filters_service.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/domain_blocked_controller_client.h"
+#include "brave/components/brave_shields/browser/domain_blocked_page.h"
+#include "brave/components/brave_shields/browser/domain_blocked_tab_storage.h"
+#include "components/prefs/pref_service.h"
+#include "components/security_interstitials/content/security_interstitial_tab_helper.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_contents_user_data.h"
+#include "net/base/net_errors.h"
+
+namespace {
+
+bool ShouldBlockDomainOnTaskRunner(
+    brave_shields::AdBlockService* ad_block_service,
+    const GURL& url) {
+  if (!ad_block_service->IsInitialized())
+    return false;
+  bool did_match_exception = false;
+  bool did_match_rule = false;
+  bool did_match_important = false;
+  std::string mock_data_url;
+  ad_block_service->ShouldStartRequest(
+      url, blink::mojom::ResourceType::kSubFrame, url.host(), &did_match_rule,
```

Comment:
I've added support for top-level document blocking filters in `adblock-rust` in [this PR](https://github.com/brave/adblock-rust/pull/155). 

Note that you'll need to pass either `"document"` or `"main_frame"` (both are equivalent) as the resource type here, rather than `kSubFrame`.

Reply:
Updated to match.